Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/AliasedEnumPropertyDrawer.cs

```diff
@@ -0,0 +1,72 @@
+#if UNITY_EDITOR
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using UnityEditor;
+using UnityEngine.UIElements;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    /// <summary>
+    /// Abstract base class for a generic property drawer for aliased enums.
+    /// </summary>
+    internal abstract class AliasedEnumPropertyDrawer<T> : PropertyDrawer where T : Enum
+    {
+        private string[] m_EnumDisplayNames;
+
+        public override VisualElement CreatePropertyGUI(SerializedProperty property)
+        {
+            ProcessDisplayNamesForAliasedEnums();
+            return base.CreatePropertyGUI(property);
+        }
+
+        protected abstract bool TryGetNonAliasedNames(string enumName, string displayName, out string outputName);
+
+        private void ProcessDisplayNamesForAliasedEnums()
+        {
+            var enumNamesAndValues = new Dictionary<string, int>();
```

Comment:
Since the enum is small and we want consistency, maybe skip this base class and just move into specific GamepadButtonPropertyDrawer class and just use a big switch statement to get it right? That would be quite straightforward. Otherwise there needs to be some smartness to get the order of each aliased combination separated by slashes consistent.